invoice/src/main/java/org/killbill/billing/invoice/InvoiceDispatcher.java

```diff
@@ -854,9 +829,23 @@ private CallContext buildCallContext(final InternalCallContext context) {
         return internalCallContextFactory.createCallContext(context);
     }
 
-    private void setChargedThroughDates(final Collection<InvoiceItem> fixedPriceItems,
-                                        final Collection<InvoiceItem> recurringItems,
-                                        final InternalCallContext context) throws SubscriptionBaseApiException {
+    private void setChargedThroughDates(final Invoice invoice, final InternalCallContext context) throws SubscriptionBaseApiException {
```

Comment:
Independently of your change, i am wondering if we should not also handle the `USAGE` types : Don't we want to end up with a non null CTD if we only bill for usage ? Might be worth creating a bug to track this-- understand what we do now, and whether this needs to be fixed.

Reply:
Fixed in 6d5af5ec7c07aaf8bd6e01a3f548132afc396a66.